queries/spark_benchmark.py

```diff
@@ -0,0 +1,238 @@
+   df.select("`id.orig_h`").count()
+  
+  #over each individual parquet instead of the merged file
+  #for root, dirs, files in os.walk(parquet_path, topdown = False):
+   #   for name in files:
+    #      df = spark.read.parquet(os.path.join(root, name))
+     #     if has_column(df, "`id.orig_h`"):
+      #        df.select("`id.orig_h`").count()
+
+def search(df, iter_num):
+   origin = sys.stdout
+   df2 = spark.sql("""SELECT *
+    FROM MERGED
+    WHERE `id.orig_h` = '10.128.0.19'
+    ORDER BY ts
+    LIMIT 5
+    """)
+   sys.stdout = open('/dev/null', 'w')
+   df2.show()
+   sys.stdout = origin
+
+def discovery(df, iter_num):
+    for root, dirs, files in os.walk(parquet_path, topdown = False):
+        for name in files:
+           # print(name)
+            df = spark.read.parquet(os.path.join(root, name))
+            df.createOrReplaceTempView("Schema")
+            df2 = spark.sql("""SELECT COUNT(*) FROM Schema""")
+            df2.count()
+    
+def path(df, iter_num):
+    origin = sys.stdout
+    df2 = spark.sql("""SELECT * FROM MERGED
+     WHERE _path = 'smb*' or _path = 'dce_rpc' """)
+    sys.stdout = open('/dev/null', 'w')
+    df2.show()
+    sys.stdout = origin
+
+def post(df, iter_num):
+    origin = sys.stdout
+    df2 = spark.sql("""SELECT ts, uid, id, method, uri, status_code
+                    from MERGED
+                    WHERE method = 'POST'
+                    """)
+    sys.stdout = open('/dev/null', 'w')
+    df2.show()
+    sys.stdout = origin
+ 
+def file_not_null(df, iter_num):
+    origin = sys.stdout
+    df2 = spark.sql("""SELECT _path, tx_hosts, rx_hosts, conn_uids, mime_type, filename, md5, sha1
+                    from MERGED
+                    WHERE filename IS NOT NULL
+                    """)
+    sys.stdout = open('/dev/null', 'w')
+    df2.show()
+    sys.stdout = origin
+    
+def count_path(df, iter_num):
+    origin = sys.stdout
+    df2 = spark.sql("""SELECT _path, count(*)
+                    from MERGED
+                    GROUP BY _path
+                    ORDER BY COUNT(*) DESC""")
+    sys.stdout = open('/dev/null', 'w')
+    df2.show()
+    sys.stdout = origin
+    
+def path_dns(df, iter_num):
+    origin = sys.stdout
+    df5 = spark.sql("""SELECT query, count(*)
+                    from MERGED
+                    WHERE _path = 'dns'
+                    GROUP BY query
+                    ORDER BY COUNT(*) DESC""")
+    sys.stdout = open('/dev/null', 'w')
+    df5.show()
+    sys.stdout = origin
+
+def http_reqs(df, iter_num):
+    origin = sys.stdout
+    df2 = spark.sql("""SELECT DISTINCT `id.orig_h`, `id.resp_h`, `id.resp_p`, method, host, uri
+                    FROM MERGED
+                    WHERE _path = 'http'
+                    """)
+    sys.stdout = open('/dev/null', 'w')
+    df2.show()
+    sys.stdout = origin
+
+def path_conn(df, iter_num):
+    origin = sys.stdout
+    df2 = spark.sql("""SELECT DISTINCT `id.orig_h`, `id.resp_h`, `id.resp_h`
```

Comment:
I think one of these `id.resp_h` should be `id.resp_p`?

Reply:
Also, can you add a sort here, to make it equivalent to the Z query? E.g., sort first by `id.orig_h`, then by `id.resp_p`, then by `id.resp_h`.